java-frontend/src/main/java/org/sonar/java/se/ProgramState.java

```diff
@@ -196,20 +186,21 @@ public boolean equals(Object o) {
     return Objects.equals(values, that.values) &&
       Objects.equals(constraints, that.constraints) &&
       Objects.equals(exitSymbolicValue, that.exitSymbolicValue) &&
-      Objects.equals(stack, that.stack);
+      Objects.equals(stack, that.stack) &&
+      Objects.equals(lastAssociatedSymbol, that.lastAssociatedSymbol);
   }
 
   @Override
   public int hashCode() {
     if (hashCode == 0) {
-      hashCode = Objects.hash(values, constraints, peekValue());
+      hashCode = Objects.hash(values, constraints, peekValue(), lastAssociatedSymbol);
```

Comment:
Same remarks as for equals. I doubt about the interest of having the last associated symbols taken into account here. LastEvaluated symbol was not part of it.